### ISSUE TITLE
INF-1 — Non-destructive scaffold of `packages/` + coexistence baseline

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1550]
**Depends on:** Phase 0 gate closed · **Blocks:** everything in Phase 1+

**Context:** The v1 rewrite is built as a new stack under `packages/` **alongside** the legacy `mace/` package, which stays **byte-frozen** on `develop` and serves as a **live, in-process differential oracle**: the parity harness (PAR-1 (#1572), in `tests/parity/`) loads both stacks in the same process and compares them numerically on every migrated capability, against the Phase 0 goldens (`tests/golden/`). Legacy is retired capability-by-capability at the very end through deletion-only PRs; the 1.0 tag is gated on `git ls-files mace/ == 0` — today that number is **83**. Until then `develop` stays installable and useful at every commit and users see zero behaviour change. The v0.3.x line is **closed**: its final release ships before this ticket starts, so no further 0.3.x release comes off `develop`. This ticket is the first move of that coexistence: it stands up the three new packages as installable-but-nearly-empty scaffolds that coexist with `mace/` in one venv and one process. It is **purely additive** — it creates files, deletes nothing, and contains no library code. In-tree adapters/shims into `mace/` are explicitly rejected as a strategy: editing the legacy package would move the numerical oracle, so the two stacks never import each other (the only planned exceptions, owned by other tickets, are the launcher's dispatch module (INF-2 (#1551)) and `tests/parity/`).

**Interface & constraints:**

1. **Three packages, fixed scopes.** Each is an independent distribution with a `src/` layout:

| Package | Import name | Scope (goes in its 2-sentence README) |
|---|---|---|
| `packages/mace-core` | `mace_core` | Framework-agnostic contract + pure math. **No torch, no jax, no e3nn** — numpy/pydantic-level dependencies only. |
| `packages/mace-torch` | `mace_torch` | The torch stack: models, training, kernel backends, deployment. Depends on `mace_core` + torch. |
| `packages/mace-jax` | `mace_jax` | The jax stack, **inference-only** — there is no JAX training anywhere in v1. Depends on `mace_core` + jax. |

2. **Per-package skeleton:** `pyproject.toml` (build backend **hatchling + hatch-vcs**, with **per-package tag prefixes** so each distribution versions independently from git tags), `src/<import_name>/`, `tests/`, `README.md` (the 2-sentence scope above). Each package exports one trivial public symbol with one passing test — enough to prove install + import + test wiring, nothing more. `requires-python` is `>=3.10`, matching the legacy dist (`setup.cfg:19`) and the four-version CI matrix (py3.10–3.13, `.github/workflows/ci-core.yaml:47`).

3. **Naming rules.** The *import* names `mace_core` / `mace_torch` / `mace_jax` are distinct from the legacy import name `mace` by construction, so both stacks import in the same process with no collision. The *distribution* names are **decided**: `mace-core`, `mace-jax` and `mace-launcher` match their import names; `packages/mace-torch` ships as **`mace-torch-v1`** while keeping the `mace_torch` import name. The legacy package holds the `mace-torch` distribution name (`setup.cfg:2`, version `attr: mace.__version__`) and two dists cannot both hold it in one venv; INF-2 (#1551) relies on this to keep console-script ownership unambiguous.

   **`mace-torch-v1` is a coexistence artifact and must never be published to PyPI.** It exists so the two stacks can be installed side by side locally. `mace-torch` on PyPI is frozen at its final v0.3.x release and stays dormant for the whole rewrite, so the name is free the moment legacy leaves the tree: **RET-6 (#1602) renames the distribution to `mace-torch`**, and REL-2 (#1604) publishes v1 under that name. PyPI has no rename operation: publishing `mace-torch-v1` even once turns the switch into a permanent user migration instead of a no-op.

4. **No console scripts.** None of the three packages declares any `console_scripts`. Entry-point ownership is INF-2 (#1551)'s job (the launcher becomes the sole owner of every `mace_*` script; two distributions declaring the same script is undefined behaviour in pip). Until INF-2 (#1551) lands, the root `setup.cfg` keeps its entry points untouched.

5. **The frozen zone is untouchable.** `mace/` (every module), the legacy `.github/workflows/` (`ci-core.yaml`, `ci-extensions.yaml`, `ci-integrations.yaml`, `ci-gpu-mpcdf.yaml`, `nightly.yaml`, `release.yaml`), the three composite actions those workflows call (`.github/actions/{run-tests,setup-mace,smoke-release}`) and the GPU pipeline definition the bridge pushes (`.github/gitlab/ci.yml`), the legacy toolchain config, and `tests/golden/` are all read-only for this ticket — the import guard is INF-3 (#1552), the dual toolchain is INF-4 (#1553). The scaffold must not create any import edge in either direction between `packages/**` and `mace` (nothing to enforce it yet; INF-3 (#1552) adds the machinery, this ticket simply must not need it).

6. **The legacy wheel must not grow a second tree.** The legacy build is `setuptools.build_meta` with an unscoped `packages = find:` (`pyproject.toml:3,6`, `setup.cfg:17-18`). Measured: `setuptools.find_packages()` from the repo root still returns exactly `['mace']` with the scaffold in place, because `find_packages` prunes directory components that are not Python identifiers and every scaffold directory is hyphenated (`mace-core`, `mace-torch`, `mace-jax`, `mace-launcher`). That is load-bearing, not incidental — a scaffold directory named with an underscore would silently ship inside the legacy wheel. Assert it rather than rely on it.

**Task:**
1. **Scaffold** `packages/{mace-core,mace-torch,mace-jax}` per the skeleton in constraints 1–2: `src/<import_name>/`, `pyproject.toml` (hatchling + hatch-vcs, per-package tag prefixes, `requires-python >=3.10`), `tests/`, 2-sentence scope README, trivial public symbol + test per package.
2. Apply the names from constraint 3 consistently in every file created here, and record in `packages/README.md` that `mace-torch-v1` is never published and becomes `mace-torch` at RET-6 (#1602).
3. **`packages/README.md`** documents the editable-install recipe for the three packages **alongside** the legacy `mace/` (fresh venv, `pip install -e` each package plus the legacy root, both stacks importable in one process).
4. Add the packaging-isolation assertion of constraint 6 to the architecture test set.
5. **Do NOT touch** `mace/`, the legacy `.github/workflows/`, `.github/actions/`, or the legacy toolchain — those are governed by INF-3 (#1552)/INF-4 (#1553).
6. Repo README transition banner: "develop = v1 rewrite in progress; `--engine=legacy` is the default and behaviour is unchanged; the v0.3.x line is closed at its final release".

**Out of scope:** launcher (INF-2 (#1551)), import-guard (INF-3 (#1552)), dual toolchain (INF-4 (#1553)), debt book (INF-5 (#1554)), any real library code (CORE-*).

**Acceptance criteria:**
- [ ] Fresh venv: `pip install -e packages/mace-core -e packages/mace-torch -e packages/mace-jax` **and** the legacy `mace/` coexist and import in the same process.
- [ ] `git ls-files mace/ | wc -l` is still **83** (nothing deleted); `git diff --stat HEAD~1 -- mace/` empty.
- [ ] `setuptools.find_packages()` at the repo root still returns exactly `['mace']` — the scaffold cannot leak into the legacy wheel.
- [ ] The full legacy suite stays green.
- [ ] The naming decision is recorded in `packages/README.md`: the import names (`mace_core`, `mace_torch`, `mace_jax`), the distribution names (`mace-core`, `mace-torch-v1`, `mace-jax`, `mace-launcher`), why `mace-torch-v1` carries the suffix, and that it is never published and becomes `mace-torch` at RET-6 (#1602).
- [ ] No package created by this ticket declares a `console_scripts` entry point; the root `setup.cfg` entry points are unchanged.

**Verify:**
```bash
python -c "import mace, mace_core, mace_torch, mace_jax; print('coexist ok')"
python -c "import setuptools; print(setuptools.find_packages())"   # ['mace']
python -m pytest packages/*/tests
python -m pytest tests/unit -m "not slow" -n auto   # legacy untouched
git ls-files mace/ | wc -l                          # 83
git diff --stat HEAD~1 -- mace/                     # empty
```
**Review focus:** that not a single line under `mace/` was touched.

---
